tests/backend/test_register.py

```diff
@@ -112,3 +112,10 @@ def test_invalid_index():
 
 def test_name_by_index():
     assert TestRegister.abi_name_by_index() == {0: "a0", 1: "a1"}
+
+
+def test_from_index():
+    assert TestRegister.from_index(0) == TestRegister.from_name("a0")
+    assert TestRegister.from_index(1) == TestRegister.from_name("a1")
```

Comment:
Would it make sense to also add `TestRegister.from_name("x0")` etc. here?

Reply:
Ah but these cannot be created from an index alone

Reply:
not sure I understand.

Reply:
Like there is no value `x` for which `TestRegister.from_index(x)` is equal to `TestRegister.from_name("x0")`

Reply:
I meant smt like

```py
assert TestRegister.from_index(0) == TestRegister.from_name("a0") == TestRegister.from_name("x0")
```

Reply:
Ah but those are different registers! `TestRegister.from_name("a0") == TestRegister(0, "a0") != TestRegister(0, "x0") == TestRegister.from_name("x0")`

Reply:
Yeah! Sorry, I meant `x10` in this case! Maybe I should lie down ;-]

Reply:
I feel like this is somewhat important to discuss at some point just so we're on the same page, maybe another day :)